odc/stats/tasks.py

```diff
@@ -238,118 +234,95 @@ def msg_default(msg):
-            elif self._frequency == "apr-oct":
-                tasks = bin_seasonal(cells, months=7, anchor=4, extract_single_season=True)
-            elif self._frequency == "annual-fy":
-                tasks = bin_seasonal(cells, months=12, anchor=7)
-            elif self._frequency == "annual":
-                tasks = bin_annual(cells)
-            elif temporal_range is not None:
-                tasks = bin_generic(cells, [temporal_range])
-            else:
-                tasks = bin_annual(cells)
-            # Remove duplicate source uuids.
-            # Duplicates occur when queried datasets are captured around UTC midnight
-            # and around weekly boundary
-            tasks = {k: set(dss) for k, dss in tasks.items()}
-            tasks_uuid = {k: [ds.id for ds in dss] for k, dss in tasks.items()}
-
-            all_ids = set()
-            for k, dss in tasks_uuid.items():
-                all_ids.update(dss)
-            msg(f"Total of {len(all_ids):,d} unique dataset IDs after filtering")
-
-            msg(f"Saving tasks to disk ({len(tasks)})")
-            cache.add_grid_tiles(self._grid, tasks_uuid)
-            msg(".. done")
-
-            self._write_info(tasks, msg, cells, debug)
+        if temporal_range is not None:
+            # Prune Datasets outside of temporal range (after correcting for UTC offset)
+            for cell in cells.values():
+                utc_offset = cell.utc_offset
+                cell.dss = [
+                    ds for ds in cell.dss if (ds.time + utc_offset) in temporal_range
+                ]
+
+        n_tiles = len(cells)
+        msg(f"Total of {n_tiles:,d} spatial tiles")
+
+        if self._frequency == "all":
+            tasks = bin_full_history(cells, start=dt_range.start, end=dt_range.end)
+        elif self._frequency == "semiannual":
+            tasks = bin_seasonal(cells, months=6, anchor=1)
+        elif self._frequency == "seasonal":
+            tasks = bin_seasonal(cells, months=3, anchor=12)
+        elif self._frequency == "nov-mar":
+            tasks = bin_seasonal(cells, months=5, anchor=11, extract_single_season=True)
+        elif self._frequency == "apr-oct":
+            tasks = bin_seasonal(cells, months=7, anchor=4, extract_single_season=True)
+        elif self._frequency == "annual-fy":
+            tasks = bin_seasonal(cells, months=12, anchor=7)
+        elif self._frequency == "annual":
+            tasks = bin_annual(cells)
+        elif temporal_range is not None:
+            tasks = bin_generic(cells, [temporal_range])
+        else:
+            tasks = bin_annual(cells)
+        # Remove duplicate source uuids.
+        # Duplicates occur when queried datasets are captured around UTC midnight
+        # and around weekly boundary
+        tasks = {k: set(dss) for k, dss in tasks.items()}
+        tasks_uuid = {k: [ds.id for ds in dss] for k, dss in tasks.items()}
```

Comment:
Could you just use `list(dss)` in place of `[ds.id for ds in dss]`?

Reply:
Dataset has many things, but here is to save only id of datasets in the task uuid. I guess it was for publishing to sqs. Though we don't do that anymore.